File: InterviewBit/Array/NobleInteger/NobleInteger.py
class Solution:
	# @param A : list of integers
	# @return an integer
	def solve(self, A):
        A.sort()
        n = len(A)
        if A[n-1]==0:
            return 1
        for i in range(n-1):
            if n-1-i== A[i] and A[i]!=A[i+1] :
                return 1
        return -1

File: InterviewBit/Array/NobleInteger/NobleInteger.cpp
int Solution::solve(vector<int> &A) {
    int size = A.size();
    sort(A.begin(), A.end());
    for (int i=0;i<size;i++){
        if (i<size-1&&A[i]==A[i+1]) continue;
        if (A[i]==size-1-i) return 1;
    }
    return -1;
}
